RELEASE_PROCEDURE.md

```diff
@@ -0,0 +1,36 @@
+# Release procedure for DLA-Future
+
+DLA-Future follows [Semantic Versioning](https://semver.org).
+
+1. For minor and major releases: create and check out a new branch at an
+   appropriate point on `main` with the name `release-major.minor`.  `major` and `minor` should be the
+   major and minor versions of the release. For patch releases: check out the corresponding
+   `release-major.minor` branch.
+
+1. Write release notes in `CHANGELOG.md`. Check for issues and pull requests for the release on the
```

Comment:
```suggestion
2. Write release notes in `CHANGELOG.md`. Check for issues and pull requests for the release on the
```
? Same for the other points.

Reply:
I know it looks funny, but this is actually rendered with correctly numbered items on GitHub (https://github.com/eth-cscs/DLA-Future/blob/369055535bd658c2151ccbba1a63e27e45c55f0b/RELEASE_PROCEDURE.md). I prefer this to explicitly having to renumber items if some are deleted/added. But then I'm not sure where this is most likely to be viewed (in an editor or rendered).

Reply:
That's neat!